### PR TITLE
include most POIs for some classes by using a blacklist

### PR DESCRIPTION
**Pull request by remi-dupre**
Include all POIs by default for OSM keys listed bellow. For `amenity` and `leisure` some specific values are still filtered out.

For all the categories this increases the number of POIs by less than 10%.

The stats below are generated using [this ugly script](https://gist.github.com/remi-dupre/94761202426a8fe46656a5d61c56ef5b#file-stats-py) and using [taginfo.openstreetmap.org](https://taginfo.openstreetmap.org).

#### amenity

 - new total: 14558376
 - previously: 13386422 (92%)
 - top 10 new imports:
   * atm: 172018 (1.18%)
   * hunting_stand: 143932 (0.99%)
   * fountain: 120275 (0.83%)
   * waste_disposal: 102683 (0.71%)
   * social_facility: 98648 (0.68%)
   * car_wash: 76726 (0.53%)
   * charging_station: 38642 (0.27%)
   * car_rental: 21729 (0.15%)
   * childcare: 20944 (0.14%)
   * driving_school: 19599 (0.13%)

#### craft

 - new total: 154164
 - previously: 147423 (96%)
 - top 10 new imports:
   * cleaning: 269 (0.17%)
   * electronics: 235 (0.15%)
   * construction: 184 (0.12%)
   * weldeur: 174 (0.11%)
   * computer: 147 (0.10%)
   * furniture: 136 (0.09%)
   * metal_works: 105 (0.07%)
   * printing: 104 (0.07%)
   * cheese: 101 (0.07%)
   * catering: 99 (0.06%)

#### leisure

 - new total: 4569962
 - previously: 4213529 (92%)
 - top 10 new imports:
   * nature_reserve: 95776 (2.10%)
   * fitness_centre: 44066 (0.96%)
   * slipway: 37320 (0.82%)
   * fitness_station: 27055 (0.59%)
   * recreation_ground: 23588 (0.52%)
   * firepit: 16524 (0.36%)
   * outdoor_seating: 11993 (0.26%)
   * sauna: 11971 (0.26%)
   * horse_riding: 10902 (0.24%)
   * resort: 10206 (0.22%)

#### shop

 - new total: 4170315
 - previously: 3700377 (89%)
 - top 10 new imports:
   * car_parts: 57980 (1.39%)
   * variety_store: 46061 (1.10%)
   * tyres: 23107 (0.55%)
   * vacant: 22589 (0.54%)
   * funeral_directors: 18819 (0.45%)
   * seafood: 15038 (0.36%)
   * houseware: 12473 (0.30%)
   * trade: 12121 (0.29%)
   * storage_rental: 11171 (0.27%)
   * pastry: 11080 (0.27%)

#### sport

 - new total: 1824567
 - previously: 1732251 (95%)
 - top 10 new imports:
   * fitness: 19141 (1.05%)
   * football: 3228 (0.18%)
   * padel: 2874 (0.16%)
   * softball: 2699 (0.15%)
   * 10pin: 2397 (0.13%)
   * futsal: 2389 (0.13%)
   * exercise: 2139 (0.12%)
   * soccer;basketball: 2016 (0.11%)
   * ice_skating: 1562 (0.09%)
   * trampoline: 1323 (0.07%)

#### tourism

 - new total: 2248230
 - previously: 2163516 (96%)
 - top 10 new imports:
   * apartment: 45691 (2.03%)
   * camp_pitch: 22621 (1.01%)
   * wilderness_hut: 8595 (0.38%)
   * trail_riding_station: 929 (0.04%)
   * heritage=6: 548 (0.02%)
   * wine_cellar: 531 (0.02%)
   * resort: 521 (0.02%)
   * cabin: 281 (0.01%)
   * disused_camp_site: 199 (0.01%)
   * winery: 166 (0.01%)

### CODE REVIEW

**Comment by remi-dupre** [comment]
I ran [another ugly script](https://gist.github.com/remi-dupre/94761202426a8fe46656a5d61c56ef5b#file-class_changes-py) to track POIs which have a new class/subclass after this new mapping. Over all POIs of Bretagne the diff seems quite small:

 -   16x ('shop', 'butcher') -> ('craft', 'caterer')
 -   16x ('sport', 'equestrian') -> ('leisure', 'horse_riding')
 -   14x ('tourism', 'attraction') -> ('amenity', 'fountain')
 -   10x ('shop', 'bakery') -> ('craft', 'bakery')
 -    5x ('shop', 'supermarket') -> ('shop', 'convenience')
 -    4x ('tourism', 'artwork') -> ('amenity', 'fountain')
 -    4x ('shop', 'jewelry') -> ('craft', 'jeweller')
 -    4x ('tourism', 'viewpoint') -> ('leisure', 'bird_hide')
 -    3x ('barrier', 'cycle_barrier') -> ('barrier', 'bollard')
 -    3x ('sport', 'judo') -> ('amenity', 'dojo')
 -    3x ('shop', 'clothes') -> ('craft', 'dressmaker')
 -    3x ('amenity', 'cafe') -> ('amenity', 'restaurant')
 -    3x ('shop', 'art') -> ('craft', 'sculptor')
 -    3x ('highway', 'bus_stop') -> ('amenity', 'shelter')
 -    2x ('sport', 'table_tennis') -> ('leisure', 'table_tennis_table')
 -    2x ('barrier', 'bollard') -> ('barrier', 'cycle_barrier')
 -    2x ('tourism', 'attraction') -> ('leisure', 'carousel')
 -    2x ('shop', 'clothes') -> ('shop', 'jewelry')
 -    2x ('tourism', 'hotel') -> ('amenity', 'restaurant;bar')
 -    2x ('amenity', 'pub') -> ('amenity', 'restaurant')
 -    2x ('shop', 'shoes') -> ('craft', 'shoemaker')
 -    2x ('sport', 'yoga') -> ('leisure', 'fitness_centre')
 -    2x ('tourism', 'attraction') -> ('amenity', 'lavoir')
 -    2x ('historic', 'monument') -> ('amenity', 'fountain')
 -    2x ('amenity', 'cafe') -> ('amenity', 'bar')
 -    2x ('shop', 'bicycle') -> ('amenity', 'bicycle_repair_station')
 -    2x ('shop', 'bed') -> ('shop', 'clothes')
 -    2x ('shop', 'shoes') -> ('shop', 'clothes')
 -    2x ('shop', 'boutique') -> ('shop', 'clothes')
 -    2x ('amenity', 'restaurant') -> ('amenity', 'fast_food')
 -    2x ('amenity', 'fast_food') -> ('amenity', 'restaurant')
 -    2x ('shop', 'convenience') -> ('craft', 'caterer')
 -    2x ('shop', 'optician') -> ('craft', 'optician')
 -    2x ('shop', 'alcohol') -> ('craft', 'brewery')
 -    2x ('shop', 'chemist') -> ('shop', 'cosmetics')
 -    1x ('tourism', 'information') -> ('amenity', 'clock')
 -    1x ('shop', 'newsagent') -> ('shop', 'tobacco')
 -    1x ('sport', 'sailing') -> ('amenity', 'boat_rental')
 -    1x ('shop', 'sports') -> ('leisure', 'fitness_centre')
 -    1x ('shop', 'laundry') -> ('craft', 'dressmaker')
 -    1x ('shop', 'supermarket') -> ('shop', 'wood')
 -    1x ('shop', 'art') -> ('shop', 'fashion_accessories')
 -    1x ('shop', 'alcohol') -> ('craft', 'distillery')
 -    1x ('amenity', 'bicycle_rental') -> ('shop', 'bicycle')
 -    1x ('amenity', 'pub') -> ('amenity', 'bar')
 -    1x ('shop', 'clothes') -> ('shop', 'hairdresser')
 -    1x ('shop', 'greengrocer') -> ('shop', 'convenience')
 -    1x ('amenity', 'arts_centre') -> ('amenity', 'music_school')
 -    1x ('shop', 'antiques') -> ('shop', 'vacant')
 -    1x ('shop', 'clothes') -> ('craft', 'tailor')

**Review by amatissart — COMMENTED** [review]
* Some POIs listed here should not be imported in my opinion:
  *  `amenity=fountain` especially if they have no name (maybe this should be implemented in the tile query, similarly to "garden")
  * `amenity=waste_disposal`
  * `leisure=nature_reserve` already handled by the layer "park". And they are typically too large to be visible entirely for zoom levels where POIs are rendered.
  * `shop=vacant` ("Shop that is currently not being used")  
&nbsp;


* How does `reject` filters work in Imposm ?  
What if an item contains both tags `tourism=museum` and `shop=no` ? We would expect it to be included, right ?

* The change in "class.sql"  defined in #32 about the new class `craft` need to be applied here. And maybe similar fallbacks would be useful for other tags ?

**Comment by remi-dupre** [comment]
>     * Some POIs listed here should not be imported in my opinion:

I just updated the list.

>     * The change in "class.sql"  defined in #32 about the new class `craft` need to be applied here. And maybe similar fallbacks would be useful for other tags ?

Nice catch, I applied this modification for both shops and crafts.

For others I didn't change anything as we were currently using the subclass as a fallback. Maybe we could use mapping key for all categories while respecting some kind of "back compatibility rules", for example ensure that all categories mapped to an icon in the style are preserved.

File: layers/poi/class.sql [hunk 1]
```diff
@@ -63,8 +62,7 @@ RETURNS TEXT AS $$
         WHEN subclass IN ('bag','clothes') THEN 'clothing_store'
         WHEN subclass IN ('swimming_area','swimming') THEN 'swimming'
         WHEN subclass IN ('castle','ruins') THEN 'castle'
-        WHEN mapping_key IN ('barrier') THEN 'barrier'
-        WHEN mapping_key IN ('shop') THEN 'shop'
+        WHEN mapping_key IN ('craft', 'shop') THEN mapping_key
```
Comment (amatissart) [inline layers/poi/class.sql]:
```suggestion
        WHEN mapping_key IN ('barrier', 'craft', 'shop') THEN mapping_key
```

File: layers/poi/layer.sql [hunk 2]
```diff
@@ -105,6 +105,24 @@ AS $$
...
+        subclass NOT IN ('yes', 'Yes', 'no', 'No', 'none')
+        AND CASE
+            WHEN mapping_key = 'amenity' THEN
+                subclass NOT IN (
+                    'bench', 'drinking_water', 'fountain', 'parking_entrance',
+                    'parking_space', 'vending_machine', 'waste_disposal',
+                    'water_point'
+                )
+            WHEN mapping_key = 'craft' THEN
+                subclass NOT IN ('vacant')
```
Comment (amatissart) [inline layers/poi/layer.sql]:
```suggestion
            WHEN mapping_key = 'shop' THEN
                subclass NOT IN ('vacant')
```

**Comment by remi-dupre** [comment]
After running the import for Ile de France using master and this PR, we don't loose any POI, but some of them or not in the same category anymore:

 -  370x ('sport', 'athletics') -> ('leisure', 'track')
 -  353x ('sport', 'swimming') -> ('leisure', 'swimming_pool')
 -   71x ('sport', 'running') -> ('leisure', 'track')
 -   45x ('sport', 'equestrian') -> ('leisure', 'horse_riding')
 -   36x ('building', 'dormitory') -> ('amenity', 'social_facility')
 -   21x ('tourism', 'artwork') -> ('amenity', 'fountain')
 -   17x ('sport', 'horse_racing') -> ('leisure', 'track')
 -   16x ('sport', 'chess') -> ('leisure', 'picnic_table')
 -   15x ('tourism', 'artwork') -> ('amenity', 'drinking_water')
 -   11x ('barrier', 'lift_gate') -> ('amenity', 'parking_entrance')
 -   11x ('barrier', 'gate') -> ('amenity', 'parking_entrance')
 -    9x ('sport', 'tennis') -> ('leisure', 'sports_hall')
 -    8x ('sport', 'cycling') -> ('leisure', 'track')
 -    8x ('landuse', 'basin') -> ('amenity', 'fountain')
 -    7x ('sport', 'table_tennis') -> ('leisure', 'table_tennis_table')
 -    7x ('sport', 'yoga') -> ('leisure', 'fitness_centre')
 -    5x ('sport', 'karting') -> ('leisure', 'track')
 -    5x ('leisure', 'sports_centre') -> ('amenity', 'gym')
 -    5x ('leisure', 'sports_centre') -> ('amenity', 'dojo')
 -    5x ('sport', 'judo') -> ('amenity', 'dojo')
 -    5x ('tourism', 'information') -> ('amenity', 'clock')
 -    4x ('sport', 'equestrian') -> ('leisure', 'track')
 -    4x ('tourism', 'information') -> ('amenity', 'vending_machine')
 -    4x ('shop', 'jewelry') -> ('craft', 'watchmaker')
 -    3x ('sport', 'soccer') -> ('leisure', 'track')
 -    3x ('sport', 'bmx') -> ('leisure', 'track')
 -    3x ('sport', 'gymnastics') -> ('leisure', 'sports_hall')
 -    3x ('shop', 'butcher') -> ('craft', 'caterer')
 -    3x ('shop', 'alcohol') -> ('craft', 'brewery')
 -    3x ('shop', 'deli') -> ('craft', 'caterer')
 -    3x ('shop', 'bicycle') -> ('amenity', 'bicycle_repair_station')
 -    3x ('shop', 'tailor') -> ('craft', 'dressmaker')
 -    2x ('sport', 'dog_racing') -> ('leisure', 'track')
 -    2x ('sport', 'motor') -> ('leisure', 'track')
 -    2x ('historic', 'castle') -> ('amenity', 'social_facility')
 -    2x ('tourism', 'information') -> ('amenity', 'reception_desk')
 -    2x ('sport', 'multi') -> ('leisure', 'fitness_centre')
 -    2x ('sport', 'multi') -> ('leisure', 'sports_hall')
 -    2x ('building', 'dormitory') -> ('amenity', 'student_accomodation')
 -    2x ('sport', 'skating') -> ('leisure', 'track')
 -    2x ('shop', 'beauty') -> ('leisure', 'tanning_salon')
 -    2x ('sport', 'boxing') -> ('leisure', 'fitness_centre')
 -    2x ('sport', 'skateboard') -> ('leisure', 'track')
 -    2x ('sport', 'scuba_diving') -> ('amenity', 'dive_centre')
 -    2x ('highway', 'bus_stop') -> ('amenity', 'vending_machine')
 -    2x ('shop', 'bakery') -> ('craft', 'bakery')
 -    2x ('shop', 'bakery') -> ('craft', 'confectionery')
 -    2x ('shop', 'convenience') -> ('craft', 'caterer')
 -    2x ('shop', 'optician') -> ('craft', 'optician')
 -    2x ('leisure', 'hackerspace') -> ('amenity', 'fablab')

**Review by amatissart — APPROVED** [review]
Looks good to me. Nice to see all these new POIs on the map!

In very dense areas (eg train stations) there are some more POIs that we could probably filter out too, such as `amenity=photo_booth` or `amenity=clock`.
But this could probably be done in another PR.

**Pull request merged**